src/shared/particles/base_particles.hpp

```diff
@@ -35,400 +35,369 @@
+template <typename DataType>
+StdLargeVec<DataType> *BaseParticles::registerSharedVariable(const std::string &variable_name)
+{
 
-        return nullptr;
+    DiscreteVariable<DataType> *variable = findVariableByName<DataType>(all_discrete_variables_, variable_name);
+
+    constexpr int type_index = DataTypeIndex<DataType>::value;
+    if (variable == nullptr)
+    {
+        UniquePtrsKeeper<StdLargeVec<DataType>> &container = std::get<type_index>(shared_particle_data_ptrs_);
+        StdLargeVec<DataType> *contained_data = container.template createPtr<StdLargeVec<DataType>>();
+        registerVariable(*contained_data, variable_name);
+        return contained_data;
     }
-    //=================================================================================================//
-    template <typename VariableType>
-    StdLargeVec<VariableType> *BaseParticles::registerSharedVariable(const std::string &variable_name)
+    else
     {
-        constexpr int type_index = DataTypeIndex<VariableType>::value;
-        if (all_variable_maps_[type_index].find(variable_name) == all_variable_maps_[type_index].end())
-        {
-            UniquePtrsKeeper<StdLargeVec<VariableType>> &container = std::get<type_index>(shared_variable_data_);
-            StdLargeVec<VariableType> *contained_data = container.template createPtr<StdLargeVec<VariableType>>();
-            registerVariable(*contained_data, variable_name);
-            return contained_data;
-        }
-        else
-        {
-            return std::get<type_index>(all_particle_data_)[all_variable_maps_[type_index][variable_name]];
-        }
+        return std::get<type_index>(all_particle_data_)[variable->IndexInContainer()];
     }
-    //=================================================================================================//
-    template <typename VariableType>
-    StdLargeVec<VariableType> *BaseParticles::getVariableByName(const std::string &variable_name)
+}
+//=================================================================================================//
+template <typename DataType>
+StdLargeVec<DataType> *BaseParticles::getVariableByName(const std::string &variable_name)
+{
+    DiscreteVariable<DataType> *variable = findVariableByName<DataType>(all_discrete_variables_, variable_name);
+
+    if (variable != nullptr)
     {
-        constexpr int type_index = DataTypeIndex<VariableType>::value;
+        constexpr int type_index = DataTypeIndex<DataType>::value;
+        return std::get<type_index>(all_particle_data_)[variable->IndexInContainer()];
+    }
 
-        if (all_variable_maps_[type_index].find(variable_name) != all_variable_maps_[type_index].end())
-            return std::get<type_index>(all_particle_data_)[all_variable_maps_[type_index][variable_name]];
+    std::cout << "\nErr0r: the variable '" << variable_name << "' is not registered!\n";
+    std::cout << __FILE__ << ':' << __LINE__ << std::endl;
+    exit(1);
```

Comment:
```suggestion
```

Reply:
we need to exit the code if the error happens.

Reply:
No, it is not needed. You are forcefully crashing the program that someone could easily recover from. It is sufficient to return a nullptr like it was before.

Example use-case:

```cpp
if(auto* ptr = getGlobalVariableByName<Vec3d>("Position"))
{
   // do something with ptr
}
```
I updated the suggestions accordingly to restore previous behavior